kong/plugins/datadog/handler.lua

```diff
@@ -43,53 +48,60 @@ local function compose_tags(service_name, status, consumer_id, tags, conf)
-                    "%.", "_")
-
-  local stat_name  = {
-    request_size     = "request.size",
-    response_size    = "response.size",
-    latency          = "latency",
-    upstream_latency = "upstream_latency",
-    kong_latency     = "kong_latency",
-    request_count    = "request.count",
-  }
-  local stat_value = {
-    request_size     = message.request and message.request.size,
-    response_size    = message.response and message.response.size,
-    latency          = message.latencies.request,
-    upstream_latency = message.latencies.proxy,
-    kong_latency     = message.latencies.kong,
-    request_count    = 1,
-  }
 
+local function log(conf, messages)
   local logger, err = statsd_logger:new(conf)
   if err then
     kong.log.err("failed to create Statsd logger: ", err)
     return
   end
 
-  for _, metric_config in pairs(conf.metrics) do
-    local stat_name       = stat_name[metric_config.name]
-    local stat_value      = stat_value[metric_config.name]
-    local get_consumer_id = get_consumer_id[metric_config.consumer_identifier]
-    local consumer_id     = get_consumer_id and get_consumer_id(message.consumer) or nil
-    local tags            = compose_tags(
-            name, message.response and message.response.status or "-",
-            consumer_id, metric_config.tags, conf)
+  for _, message in ipairs(messages) do
+    local name = gsub(message.service.name ~= null and
+                      message.service.name or message.service.host,
+                      "%.", "_")
+
+    local stat_name  = {
+      request_size     = "request.size",
+      response_size    = "response.size",
+      latency          = "latency",
+      upstream_latency = "upstream_latency",
+      kong_latency     = "kong_latency",
+      request_count    = "request.count",
+    }
+    local stat_value = {
+      request_size     = message.request and message.request.size,
+      response_size    = message.response and message.response.size,
+      latency          = message.latencies.request,
+      upstream_latency = message.latencies.proxy,
+      kong_latency     = message.latencies.kong,
+      request_count    = 1,
+    }
+
+    for _, metric_config in pairs(conf.metrics) do
+      local stat_name       = stat_name[metric_config.name]
+      if stat_name == nil then
+        goto continue
+      end
+
+      local stat_value      = stat_value[metric_config.name]
+      local get_consumer_id = get_consumer_id[metric_config.consumer_identifier]
+      local consumer_id     = get_consumer_id and get_consumer_id(message.consumer) or nil
+      local tags            = compose_tags(
+              name, message.response and message.response.status or "-",
+              consumer_id, metric_config.tags, conf)
 
-    if stat_name ~= nil then
       logger:send_statsd(stat_name, stat_value,
                          logger.stat_types[metric_config.stat_type],
                          metric_config.sample_rate, tags)
     end
+    ::continue::
```

Comment:
The tag `::continue::` seems to be the wrong place.

It should be in the `for _, metric_config` loop, i.e, move up one line.

Reply:
ah yes, my mistake